more_itertools/more.py

```diff
@@ -1411,6 +1413,41 @@ def stagger(iterable, offsets=(-1, 0, 1), longest=False, fillvalue=None):
     )
 
 
+class UnequalIterablesException(Exception):
+    def __init__(self, idxs, names=None):
+        if names is None:
+            depleted = "Indices: {}".format(
+                ", ".join(str(idx) for idx in idxs)
+            )
+        else:
+            depleted = "Columns: {}".format(
+                ", ".join(names[idx] for idx in idxs)
+            )
+        super().__init__(
+            "Iterables have different lengths."
+            "{} depleted before rest.".format(depleted)
+        )
+
+
+def zip_equal(*iterables, names=None):
+    """``zip`` the input *iterables* together, but throw an
+    ``UnequalIterablesException`` if any of the *iterables* terminate before
+    the others. The exception will give the indices which terminated too early.
+    Alternatively *names* can be passed and it will give the name corresponding
+    to the early terminating indices.
+
+    """
+    sentinel = object()
+    if names is not None and len(iterables) != len(names):
+        raise ValueError("Number of iterable and names didn't match")
+
+    for combo in zip_longest(*iterables, fillvalue=sentinel):
+        finished_idxs = [idx for idx, c in enumerate(combo) if c is sentinel]
+        if finished_idxs:
+            raise UnequalIterablesException(finished_idxs, names)
```

Comment:
I'm OK with this being its own `Exception` class, but I think it might be overengineered right now. How about just:

```python
class UnequalIterablesError(ValueError):
    pass
```

Reply:
`Error` seems to be more of a standard suffix than `Exception` - see [here](https://docs.python.org/3/library/exceptions.html#exception-hierarchy).